lib/bundler/dsl.rb

```diff
@@ -46,7 +44,7 @@ def eval_gemfile(gemfile, contents = nil)
 
       raise DSLError.new(message, gemfile, e.backtrace, contents)
     ensure
-      @gemfile = original_gemfile
+      @gemfile = gemfile
```

Comment:
this is incorrect, it needs to reference the value of `@gemfile` before the method was invoked